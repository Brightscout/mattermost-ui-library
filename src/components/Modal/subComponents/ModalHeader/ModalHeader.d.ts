export interface ModalHeaderProps {

    /**
     * Main title of the header
     */
    title: string;

    /**
     * Subtitle of the header
     */
    subtitle?: string;

    /**
     * Callback to be triggered on clicking the close icon button
     */
    onHeaderCloseHandler: () => void;
}
