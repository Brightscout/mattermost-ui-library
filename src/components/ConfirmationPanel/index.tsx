import React, {forwardRef} from 'react';

// Components
import ModalBody from '@Components/Modal/subComponents/ModalBody';
import ModalFooter from '@Components/Modal/subComponents/ModalFooter';
import ModalSubtitleAndError from '@Components/Modal/subComponents/ModalSubtitleAndError';

type ConfirmationPanelProps = {
    confirmationMsg: string;
    onHide: () => void;
    onConfirm?: (() => void) | null;
    loading?: boolean;
    className?: string;
    error?: string | JSX.Element;
}

const ConfirmationPanel = forwardRef<HTMLDivElement, ConfirmationPanelProps>(({
    confirmationMsg,
    onHide,
    onConfirm,
    loading,
    className = '',
    error,
}: ConfirmationPanelProps, confirmationPanelRef) => (
    <div
        ref={confirmationPanelRef}
        className={`confirmation-panel primary-panel ${className}`}
    >
        <ModalBody>
            <p className='margin-top-10 margin-h-0 margin-bottom-15'>{confirmationMsg}</p>
        </ModalBody>
        <ModalSubtitleAndError error={error}/>
        <ModalFooter
            onConfirm={onConfirm}
            onHide={onHide}
            confirmBtnText='Delete'
            cancelBtnText='Cancel'
            confirmDisabled={loading}
            cancelDisabled={loading}
            className='padding-v-12 padding-h-15'
        />
    </div>
));

export default ConfirmationPanel;
