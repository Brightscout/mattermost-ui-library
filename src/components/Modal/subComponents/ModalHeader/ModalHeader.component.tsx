import React from 'react';

import {Icon} from '@Components/Icon';

import {ModalHeaderProps} from './ModalHeader';
import {
    IconButton,
    ModalHeaderRow,
    ModalHeaderSubtitle,
    ModalHeaderTitle,
    ModalHeaderWrapper,
} from './ModalHeader.styles';

/**
 * ModalHeader Component
 *
 * @example
 * ```ts
 * <ModalHeader
 *      title='Modal Header'
        subtitle='Modal Subtitle'
        onCloseHandler={onHideHandler}
 * />
 * ```
 */
export const ModalHeader = (props: ModalHeaderProps) => {
    const {title, subtitle, onCloseHandler} = props;
    return (
        <ModalHeaderWrapper className='mm-modalHeader'>
            <ModalHeaderRow>
                <ModalHeaderTitle className='mm-modalHeaderTitle'>
                    {title}
                </ModalHeaderTitle>
                <IconButton onClick={onCloseHandler}>
                    <Icon
                        name='Close'
                        size={20}
                    />
                </IconButton>
            </ModalHeaderRow>
            {subtitle && (
                <ModalHeaderSubtitle className='mm-modalHeaderSubtitle'>
                    {subtitle}
                </ModalHeaderSubtitle>
            )}
        </ModalHeaderWrapper>
    );
};
