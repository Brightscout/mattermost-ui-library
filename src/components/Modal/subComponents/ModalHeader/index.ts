export {ModalHeader} from './ModalHeader.component';
