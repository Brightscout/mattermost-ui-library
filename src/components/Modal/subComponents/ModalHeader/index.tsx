import React from 'react';
import {Modal as RBModal} from 'react-bootstrap';

type ModalHeaderProps = {
    title?: string | JSX.Element;
    onHide: () => void;
    showCloseIconInHeader?: boolean;
    className?: string;
}

const ModalHeader = ({
    title,
    showCloseIconInHeader,
    onHide,
    className = '',
}: ModalHeaderProps) : JSX.Element => (
    <>{(title || showCloseIconInHeader) && (
        <RBModal.Header className={`modal__header ${className}`}>
            <div className='modal__title d-flex align-items-center justify-content-between'>
                {title && <p className='modal__title text-ellipsis font-size-20 wt-500 margin-0'>{title}</p>}
                {showCloseIconInHeader && (
                    <button
                        className='style--none'
                        onClick={onHide}
                    >
                        <i className='icon icon-close modal__close-icon cursor-pointer font-size-20'/>
                    </button>
                )}
            </div>
        </RBModal.Header>
    )}
    </>
);

export default ModalHeader;
