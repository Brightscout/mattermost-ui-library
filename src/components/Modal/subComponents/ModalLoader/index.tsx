import React from 'react';

import LinearLoader from '@Components/Loader/Linear';

type ModalLoaderProps = {
    loading?: boolean;
    className?: string,
}

const ModalLoader = ({loading, className = ''}: ModalLoaderProps): JSX.Element => (
    <div className={`modal__loader-container position-relative ${className}`}>
        {loading && <LinearLoader/>}
    </div>
);

export default ModalLoader;
