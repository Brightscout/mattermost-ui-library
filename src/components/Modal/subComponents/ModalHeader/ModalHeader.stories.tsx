import React from 'react';
import {Story, Meta} from '@storybook/react';

import {ModalHeader} from './ModalHeader.component';
import {ModalHeaderProps} from './ModalHeader';

export default {
    title: 'Component/Modal/ModalHeader',
    component: ModalHeader,
    args: {
        title: 'Modal Header',
    },
} as Meta<typeof ModalHeader>;

const ModalHeaderTemplate: Story<ModalHeaderProps> = (args) => <ModalHeader {...args}/>;

// Default
export const Default = ModalHeaderTemplate.bind({});

export const WithSubtitle = ModalHeaderTemplate.bind({});
WithSubtitle.args = {
    subtitle: 'Diam viverra donec nisi in lorem dictum',
};
