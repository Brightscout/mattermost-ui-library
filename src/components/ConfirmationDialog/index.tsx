import React from 'react';

// Components
import Modal from '@Components/Modal/customModal';
import ConfirmationModal from '@Components/ConfirmationPanel';
import ModalHeader from '@Components/Modal/subComponents/ModalHeader';
import ModalLoader from '@Components/Modal/subComponents/ModalLoader';

type ConfirmationDialogProps = {
    show: boolean;
    onHide: () => void;
    loading?: boolean;
    onConfirm: () => void;
    className?: string;
}

const ConfirmationDialog = ({
    show,
    onHide,
    loading,
    onConfirm,
    className = '',
}: ConfirmationDialogProps) => (
    <Modal
        show={show}
        onHide={onHide}
        className={`delete-confirmation-dialog ${className}`}
    >
        <>
            <ModalHeader
                title='Confirm Delete Feed'
                onHide={onHide}
                showCloseIconInHeader={true}
            />
            <ModalLoader loading={loading}/>
            <ConfirmationModal
                loading={loading}
                onHide={onHide}
                onConfirm={onConfirm}
            />
        </>
    </Modal>
);

export default ConfirmationDialog;
