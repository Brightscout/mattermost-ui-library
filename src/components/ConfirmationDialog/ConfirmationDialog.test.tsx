import React from 'react';
import {shallow, ShallowWrapper} from 'enzyme';

import ModalLoader from '@Components/Modal/subComponents/ModalLoader';
import ModalHeader from '@Components/Modal/subComponents/ModalHeader';
import ConfirmationPanel from '@Components/ConfirmationPanel';

// Mock data
import {confirmHandler, hideHandler, confirmationDialogProps} from '@Constants/mockData/confirmationDialog';

import ConfirmationDialog from '.';

describe('ConfirmationDialog', () => {
    let component: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;

    beforeEach(() => {
        component = shallow(<ConfirmationDialog {...confirmationDialogProps} />);
    });

    it('Should render correctly', () => {
        expect(component).toMatchSnapshot();
    });

    it('Should apply the passed "className" prop', () => {
        expect(component.hasClass(confirmationDialogProps.className)).toBeTruthy();
    });

    it('Should render the header correctly', () => {
        expect(component.contains(
            <ModalHeader
                title={confirmationDialogProps.title}
                onHide={hideHandler}
                showCloseIconInHeader={true}
            />,
        )).toBeTruthy();
    });

    it('Should render the confirmationPanel correctly', () => {
        expect(component.contains(
            <ConfirmationPanel
                loading={confirmationDialogProps.loading}
                onHide={hideHandler}
                onConfirm={confirmHandler}
                confirmationMsg={confirmationDialogProps.confirmationMsg}
                error={confirmationDialogProps.error}
            />,
        )).toBeTruthy();
    });

    it('Should show the loader when "loading" prop is true', () => {
        expect(component.contains(<ModalLoader loading={confirmationDialogProps.loading} />)).toBeTruthy();
    });
});
