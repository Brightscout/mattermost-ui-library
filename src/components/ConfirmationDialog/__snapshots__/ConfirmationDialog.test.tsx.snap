// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationDialog Should render correctly 1`] = `
<CustomModal
  className="delete-confirmation-dialog confirmation-dialog-class"
  onHide={[MockFunction]}
  show={true}
>
  <ModalHeader
    onHide={[MockFunction]}
    showCloseIconInHeader={true}
    title="Dialog Title"
  />
  <ModalLoader
    loading={true}
  />
  <ForwardRef
    confirmationMsg="Dialog Message"
    error="Dialog error"
    loading={true}
    onConfirm={[MockFunction]}
    onHide={[MockFunction]}
  />
</CustomModal>
`;
